### ISSUE TITLE
[Bug]: AgentState.ERROR for OpenRouter Haiku and 4o-mini

### ISSUE THREAD

**Issue by champ2050** [OpenHands/OpenHands#4841]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### What happened?

**Description**:
When running the OpenHands server (version 0.12), the following sequence occurs, ending with the agent entering `AgentState.ERROR`. The system fails to restore the agent state from a previous session, and the agent does not recover. This issue may relate to missing configuration for `GITHUB_USER_LIST_FILE` and `GITHUB_USERS_SHEET_ID`.

**Steps to Reproduce**:
1. Start the OpenHands server.
2. Observe the logs for the initialization sequence and application startup.
3. Note that the `Agent Controller` attempts to restore the state and subsequently enters the `ERROR` state.

**Expected Behavior**:
The `Agent Controller` should restore the state from the session file without issues or should provide a clear recovery path if the session file is not found.

**Actual Behavior**:
- The log shows a warning indicating failure to restore state:  
  `Failed to restore state from session: sessions/a87fa5a6-1c7b-4a62-aeec-615373c5e556/agent_state.pkl`
- The agent then proceeds to enter `AgentState.ERROR` after multiple initialization attempts.

**Environment**:
- **OpenHands version**: 0.12
- **Environment details**:
  - Docker Desktop 4.35.1
  - Windows 10 Version 22H2 (19045.5011)
  - Browser: Google Chrome
  - API Key Used: OpenAI (currently)  
  - Prior Attempted Model: OpenRouter with **Anthropic Claude Haiku 3.5** (encountered similar issue but logs not available), now using OpenAI model GPT-4o-mini

**Additional Information**:
- `GITHUB_USER_LIST_FILE` and `GITHUB_USERS_SHEET_ID` are not configured, which may be related to this issue.
- The runtime container initializes and waits for the client at `http://host.docker.internal:38461`.
- If there are required configuration settings or known workarounds for state restoration, please provide further guidance.

### OpenHands Installation

Docker command in README

### OpenHands Version

0.12

### Operating System

WSL on Windows

### Relevant log output

```shell
2024-11-08 11:15:59 
2024-11-08 11:15:59 Starting OpenHands...
2024-11-08 11:15:59 Running OpenHands as root
2024-11-08 11:16:39 INFO:     Started server process [10]
2024-11-08 11:16:39 INFO:     Waiting for application startup.
2024-11-08 11:16:39 INFO:     Application startup complete.
2024-11-08 11:16:39 INFO:     Uvicorn running on http://0.0.0.0:3000 (Press CTRL+C to quit)
2024-11-08 11:17:05 INFO:     172.17.0.1:39640 - "GET / HTTP/1.1" 200 OK
2024-11-08 11:17:05 INFO:     172.17.0.1:39640 - "GET /locales/en/translation.json HTTP/1.1" 200 OK
2024-11-08 11:17:05 INFO:     172.17.0.1:39640 - "GET /favicon.ico HTTP/1.1" 200 OK
2024-11-08 11:17:05 INFO:     172.17.0.1:39640 - "GET /config.json HTTP/1.1" 200 OK
2024-11-08 11:17:05 INFO:     172.17.0.1:39640 - "GET /config.json HTTP/1.1" 200 OK
2024-11-08 11:17:06 INFO:     172.17.0.1:39640 - "GET /api/options/models HTTP/1.1" 200 OK
2024-11-08 11:17:06 INFO:     172.17.0.1:39646 - "GET /api/options/security-analyzers HTTP/1.1" 200 OK
2024-11-08 11:17:06 INFO:     172.17.0.1:39658 - "GET /api/options/agents HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56610 - "GET / HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56610 - "GET /assets/root-C13iIuce.css HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56620 - "GET /assets/manifest-4225960c.js HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56628 - "GET /assets/components-Bd8d2nMd.js HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56626 - "GET /assets/entry.client-1n4ZPLSW.js HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56610 - "GET /assets/index-CB_aKBva.js HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56620 - "GET /assets/socket-B39-LSlI.js HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56642 - "GET /assets/react-redux-DzUTqGNF.js HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56654 - "GET /assets/jsx-runtime-zhgTtAea.js HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56626 - "GET /assets/store-CT-MXBfU.js HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56620 - "GET /assets/index-DvhLSMS6.js HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56642 - "GET /assets/router-CQIzOUUf.js HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56628 - "GET /assets/i18nInstance-DCxlOlkw.js HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56654 - "GET /assets/AgentState-DCyv5jZT.js HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56626 - "GET /assets/codeSlice-IFyzp_hl.js HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56610 - "GET /assets/agentSlice-0xeL04tr.js HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56642 - "GET /assets/index-BgBamcox.js HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56628 - "GET /assets/root-CjnzvPFH.js HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56628 - "GET /locales/en/translation.json HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56642 - "GET /favicon.ico HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56642 - "GET /config.json HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56628 - "GET /config.json HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56628 - "GET /api/options/models HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56642 - "GET /api/options/agents HTTP/1.1" 200 OK
2024-11-08 11:17:15 INFO:     172.17.0.1:56626 - "GET /api/options/security-analyzers HTTP/1.1" 200 OK
2024-11-08 11:17:23 INFO:     172.17.0.1:56670 - "GET / HTTP/1.1" 200 OK
2024-11-08 11:17:23 INFO:     172.17.0.1:56670 - "GET /locales/en/translation.json HTTP/1.1" 200 OK
2024-11-08 11:17:24 INFO:     172.17.0.1:56670 - "GET /config.json HTTP/1.1" 200 OK
2024-11-08 11:17:24 INFO:     172.17.0.1:56670 - "GET /config.json HTTP/1.1" 200 OK
2024-11-08 11:17:24 INFO:     172.17.0.1:56670 - "GET /api/options/models HTTP/1.1" 200 OK
2024-11-08 11:17:24 INFO:     172.17.0.1:33110 - "GET /api/options/agents HTTP/1.1" 200 OK
2024-11-08 11:17:24 INFO:     172.17.0.1:33122 - "GET /api/options/security-analyzers HTTP/1.1" 200 OK
2024-11-08 11:18:12 INFO:     172.17.0.1:39544 - "GET /config.json HTTP/1.1" 200 OK
2024-11-08 11:18:12 05:48:12 - openhands:INFO: github.py:14 - Initializing UserVerifier
2024-11-08 11:18:12 05:48:12 - openhands:INFO: github.py:27 - GITHUB_USER_LIST_FILE not configured
2024-11-08 11:18:12 05:48:12 - openhands:INFO: github.py:48 - GITHUB_USERS_SHEET_ID not configured
2024-11-08 11:18:12 05:48:12 - openhands:INFO: github.py:85 - No user verification sources configured - allowing all users
2024-11-08 11:18:12 INFO:     ('172.17.0.1', 39552) - "WebSocket /ws" [accepted]
2024-11-08 11:18:12 05:48:12 - openhands:INFO: listen.py:336 - New session: a87fa5a6-1c7b-4a62-aeec-615373c5e556
2024-11-08 11:18:12 INFO:     connection open
2024-11-08 11:18:13 INFO:     172.17.0.1:39544 - "GET /config.json HTTP/1.1" 200 OK
2024-11-08 11:18:13 05:48:13 - openhands:INFO: base.py:98 - [runtime a87fa5a6-1c7b-4a62-aeec-615373c5e556] Starting runtime with image: docker.all-hands.dev/all-hands-ai/runtime:0.12-nikolaik
2024-11-08 11:18:17 05:48:17 - openhands:INFO: base.py:98 - [runtime a87fa5a6-1c7b-4a62-aeec-615373c5e556] Container started: openhands-runtime-a87fa5a6-1c7b-4a62-aeec-615373c5e556
2024-11-08 11:18:17 05:48:17 - openhands:INFO: base.py:98 - [runtime a87fa5a6-1c7b-4a62-aeec-615373c5e556] Waiting for client to become ready at http://host.docker.internal:38461...
2024-11-08 11:19:32 05:49:32 - openhands:INFO: base.py:98 - [runtime a87fa5a6-1c7b-4a62-aeec-615373c5e556] Runtime is ready.
2024-11-08 11:19:33 05:49:33 - openhands:WARNING: state.py:118 - Failed to restore state from session: sessions/a87fa5a6-1c7b-4a62-aeec-615373c5e556/agent_state.pkl
2024-11-08 11:19:33 05:49:33 - openhands:INFO: agent_controller.py:135 - [Agent Controller a87fa5a6-1c7b-4a62-aeec-615373c5e556] Starting step loop...
2024-11-08 11:19:33 05:49:33 - openhands:INFO: agent_controller.py:135 - [Agent Controller a87fa5a6-1c7b-4a62-aeec-615373c5e556] Setting agent(CodeActAgent) state from AgentState.LOADING to AgentState.INIT
2024-11-08 11:19:33 05:49:33 - openhands:INFO: agent_controller.py:135 - [Agent Controller a87fa5a6-1c7b-4a62-aeec-615373c5e556] Setting agent(CodeActAgent) state from AgentState.INIT to AgentState.RUNNING
2024-11-08 11:19:33 05:49:33 - openhands:INFO: github.py:14 - Initializing UserVerifier
2024-11-08 11:19:33 05:49:33 - openhands:INFO: github.py:27 - GITHUB_USER_LIST_FILE not configured
2024-11-08 11:19:33 05:49:33 - openhands:INFO: github.py:48 - GITHUB_USERS_SHEET_ID not configured
2024-11-08 11:19:33 05:49:33 - openhands:INFO: github.py:85 - No user verification sources configured - allowing all users
2024-11-08 11:19:33 INFO:     172.17.0.1:53844 - "GET /config.json HTTP/1.1" 200 OK
2024-11-08 11:19:33 INFO:     172.17.0.1:53844 - "GET /config.json HTTP/1.1" 200 OK
2024-11-08 11:19:33 INFO:     172.17.0.1:53830 - "GET /api/list-files HTTP/1.1" 200 OK
2024-11-08 11:19:33 05:49:33 - openhands:INFO: github.py:14 - Initializing UserVerifier
2024-11-08 11:19:33 05:49:33 - openhands:INFO: github.py:27 - GITHUB_USER_LIST_FILE not configured
2024-11-08 11:19:33 05:49:33 - openhands:INFO: github.py:48 - GITHUB_USERS_SHEET_ID not configured
2024-11-08 11:19:33 05:49:33 - openhands:INFO: github.py:85 - No user verification sources configured - allowing all users
2024-11-08 11:19:33 INFO:     172.17.0.1:53830 - "GET /api/list-files HTTP/1.1" 200 OK
2024-11-08 11:19:33 05:49:33 - openhands:INFO: github.py:14 - Initializing UserVerifier
2024-11-08 11:19:33 05:49:33 - openhands:INFO: github.py:27 - GITHUB_USER_LIST_FILE not configured
2024-11-08 11:19:33 05:49:33 - openhands:INFO: github.py:48 - GITHUB_USERS_SHEET_ID not configured
2024-11-08 11:19:33 05:49:33 - openhands:INFO: github.py:85 - No user verification sources configured - allowing all users
2024-11-08 11:19:33 INFO:     172.17.0.1:53844 - "GET /api/list-files HTTP/1.1" 200 OK
2024-11-08 11:19:34 
2024-11-08 11:19:34 
2024-11-08 11:19:34 ==============
2024-11-08 11:19:34 [Agent Controller a87fa5a6-1c7b-4a62-aeec-615373c5e556] LEVEL 0 LOCAL STEP 0 GLOBAL STEP 0
2024-11-08 11:19:34 
2024-11-08 11:19:34 
2024-11-08 11:19:34 Give Feedback / Get Help: https://github.com/BerriAI/litellm/issues/new
2024-11-08 11:19:34 LiteLLM.Info: If you need to debug this error, use `litellm.set_verbose=True'.
2024-11-08 11:19:34 
2024-11-08 11:19:35 05:49:35 - openhands:INFO: agent_controller.py:135 - [Agent Controller a87fa5a6-1c7b-4a62-aeec-615373c5e556] Setting agent(CodeActAgent) state from AgentState.RUNNING to AgentState.ERROR
2024-11-08 11:19:35 05:49:35 - openhands:INFO: github.py:14 - Initializing UserVerifier
2024-11-08 11:19:35 05:49:35 - openhands:INFO: github.py:27 - GITHUB_USER_LIST_FILE not configured
2024-11-08 11:19:35 05:49:35 - openhands:INFO: github.py:48 - GITHUB_USERS_SHEET_ID not configured
2024-11-08 11:19:35 05:49:35 - openhands:INFO: github.py:85 - No user verification sources configured - allowing all users
2024-11-08 11:19:35 INFO:     172.17.0.1:53830 - "GET /config.json HTTP/1.1" 200 OK
2024-11-08 11:19:35 INFO:     172.17.0.1:53844 - "GET /api/list-files HTTP/1.1" 200 OK
```

**Comment by enyst** [comment]
Can you please add to the README command: `DEBUG=1` and then run it again and check the logs?

I think it's an LLM error that is not displayed.

The state message can be ignored. If a previous state cannot be restored, then the app initializes a new state, that's not really an error. We should improve this... it's confusing.

**Comment by enyst** [comment]
Also, it might be useful to try with Claude Sonnet 3.5 just for one run, to see if it works?

We had just these days a couple of issues solved for both Haiku and 4o-mini, I think you may have run into one...

**Comment by zhaozhiming** [comment]
I am experiencing the same problem. I am using the OpenAI GPT-4 model and have added DEBUG=1, but I don’t see any useful messages.

**Comment by enyst** [comment]
@champ2050 @zhaozhiming  We just released 0.13.0, please give it a try and let us know!

**Comment by zhaozhiming** [comment]
@enyst I tested the 0.13.0 version and no longer encountered this error. Thx.
